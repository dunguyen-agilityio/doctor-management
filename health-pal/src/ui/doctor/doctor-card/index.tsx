import LocationOutline from '@/icons/location-outline'
import { TDoctorCard } from '@/models/doctor'
import { useFavoritesStore } from '@/stores/favorite'

import { memo, useRef } from 'react'

import { Link } from 'expo-router'

import { Card, Separator, XStack, YStack } from 'tamagui'

import { ROUTES } from '@/constants'

import { useAddFavorite } from '@/hooks/use-add-favorite'
import { useRemoveFavorite } from '@/hooks/use-remove-favorite'

import { CloudinaryImage, Heading, Stars, Text } from '@/components'

import FavoriteButton from '@/ui/favorite/favorite-button'
import RemoveFavoriteModal from '@/ui/favorite/remove-favorite-confirm-modal'

import { FAVORITE_TYPES } from '@/types/favorite'
import { ModalRef } from '@/types/modal'

interface DoctorCardProps extends TDoctorCard {
  showReview?: boolean
  actionable?: boolean
}

const DoctorCard = ({
  avatar,
  name,
  specialty,
  address,
  reviewCounter,
  rating,
  showReview = true,
  id,
  actionable = true,
  documentId,
}: DoctorCardProps) => {
  const { mutate: removeFavorite, isPending: removeFavPending } = useRemoveFavorite(
    FAVORITE_TYPES.DOCTOR,
    name,
  )

  const { mutate: addFavorite, isPending: addFavPending } = useAddFavorite(
    FAVORITE_TYPES.DOCTOR,
    name,
  )

  const favoriteId = useFavoritesStore((state) => state.favoriteDoctors[id])

  const confirmRef = useRef<ModalRef>(null)

  const handleFavorite = () => {
    if (favoriteId) {
      confirmRef.current?.open()
      return
    }

    addFavorite(id)
  }

  const handleRemove = () => {
    confirmRef.current?.close()
    removeFavorite(favoriteId)
  }

  const renderCard = (action = false) => (
    <Card
      elevate
      bordered
      flexDirection="row"
      padding={12}
      paddingRight={0}
      borderWidth={0.5}
      borderRadius={12}
      position="relative"
      borderColor="$grey100"
      shadowColor="$black"
      shadowOffset={{ width: 4, height: 4 }}
      shadowOpacity={0.1}
      disabledStyle={{ opacity: 0.8 }}
      gap={12}
      shadowRadius={12}
      elevation={3}>
      <Card.Header padding={0}>
        <CloudinaryImage
          source={{ uri: avatar }}
          style={{ width: 110, height: 110, borderRadius: 12 }}
          tabIndex={0}
          aria-label={`Image of ${name}`}
          role="img"
        />
      </Card.Header>
      <Card.Footer flex={1} paddingVertical={0} marginVertical={0}>
        {showReview && (
          <FavoriteButton
            favoriteId={favoriteId}
            type={FAVORITE_TYPES.DOCTOR}
            itemId={id}
            itemName={name}
            top={0}
            right={6}
            position="absolute"
            zIndex={1000}
            {...(action && {
              onPress: handleFavorite,
              'aria-label': favoriteId
                ? `Remove ${name} from favorites`
                : `Add ${name} to favorites`,
              tabIndex: 0,
            })}
            role="button"
          />
        )}
        <YStack paddingRight={12} flex={1}>
          <XStack justifyContent="space-between" alignItems="center">
            <Heading>{name}</Heading>
          </XStack>
          <Separator marginVertical={8} />
          <YStack gap={showReview ? 0 : 8}>
            <Text size="small" fontWeight="600">
              {specialty}
            </Text>
            <XStack alignItems="center" gap="$sm" overflow="hidden">
              <LocationOutline />
              <Text size="small" width={200} numberOfLines={1} testID="address-text">
                {address}
              </Text>
            </XStack>
          </YStack>
          {showReview && (
            <XStack alignItems="center">
              <Stars stars={rating} max={1} flexDirection="row-reverse" size={15} />
              <Separator vertical marginHorizontal={8} height={13} />
              <Text size="extraSmall">{`${reviewCounter} Reviews`}</Text>
            </XStack>
          )}
        </YStack>
      </Card.Footer>
    </Card>
  )

  const content = actionable ? (
    <Link
      testID="doctor-link"
      accessibilityLabel={`View doctors at ${name}`}
      accessibilityHint="Shows this doctor in the doctor list"
      role="link"
      disabled={removeFavPending || addFavPending}
      href={{ pathname: ROUTES.DOCTOR, params: { id: documentId } }}>
      {renderCard(true)}
    </Link>
  ) : (
    renderCard(true)
  )

  return (
    <>
      {content}
      {favoriteId && (
        <RemoveFavoriteModal onConfirm={handleRemove} ref={confirmRef}>
          {renderCard()}
        </RemoveFavoriteModal>
      )}
    </>
  )
}

export default memo(DoctorCard)
