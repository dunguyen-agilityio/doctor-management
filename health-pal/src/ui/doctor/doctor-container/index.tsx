import { Spinner, YStack } from 'tamagui'

import { useDoctors } from '@/hooks/use-doctors'
import { useFavoriteDoctors } from '@/hooks/use-favorite'

import { Empty, ErrorState, LoadingIndicator, Text, XStack } from '@/components'
import DoctorListSkeleton from '@/components/skeleton/doctor-list-skeleton'

import DoctorList from '../doctor-list'

interface DoctorContainerProps {
  query: string
  specialty: string[]
}

const DoctorContainer = ({ query, specialty }: DoctorContainerProps) => {
  const {
    data,
    isFetching,
    isLoading: docLoading,
    error,
    fetchNextPage,
    refetch,
    hasNextPage,
    isFetchingNextPage,
  } = useDoctors(query, specialty)

  const { isLoading: favLoading } = useFavoriteDoctors()

  const hasError = error || !data

  const isLoading = favLoading || docLoading

  if (isLoading) {
    return <DoctorListSkeleton />
  }

  if (!data && hasError) {
    return (
      <ErrorState
        title="Failed to Load"
        message="Please check your internet connection."
        onRetry={refetch}
      />
    )
  }

  const { data: doctors } = data

  const ListFooterComponent = hasNextPage ? (
    <YStack paddingVertical={8}>
      <Spinner testID="spinner" />
    </YStack>
  ) : null

  const ListEmptyComponent = (
    <Empty
      title="No Doctors Found"
      description="Try adjusting your filters or check back later."
      actionLabel="Refresh"
      onAction={refetch}
    />
  )

  const onEndReached = () => fetchNextPage()

  return (
    <YStack flex={1} gap={8} position="relative">
      {isFetching && !isFetchingNextPage && <LoadingIndicator fullScreen />}
      <XStack justifyContent="space-between" paddingHorizontal={24}>
        <Text size="medium" fontWeight="700">{`${data?.meta.pagination.total ?? 0} founds`}</Text>
      </XStack>
      <DoctorList
        data={doctors}
        onEndReached={onEndReached}
        ListFooterComponent={ListFooterComponent}
        ListEmptyComponent={ListEmptyComponent}
      />
    </YStack>
  )
}

export default DoctorContainer
