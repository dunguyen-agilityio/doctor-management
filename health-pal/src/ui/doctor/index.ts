export { default as DoctorContainer } from './doctor-container'
export { default as DoctorDetailHeader } from './doctor-header'
export { default as MultipleSelectSpecialty } from './multiple-select-specialty'
