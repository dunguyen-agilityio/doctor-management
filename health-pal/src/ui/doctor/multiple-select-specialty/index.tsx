import { ScrollView } from 'tamagui'

import { SPECIALTY_LIST } from '@/constants'

import { Chip } from '@/components'

const MultipleSelectSpecialty = ({
  onChange,
  values: defaultValues,
}: {
  onChange: (value: string[]) => void
  values: string[]
}) => {
  const handleSelect = (value: string) => {
    let values = [...defaultValues]

    if (value === 'all') {
      values = ['all']
    } else if (!values.includes(value)) {
      values = values.filter((item) => item !== 'all')
      values.push(value)
    } else {
      values = values.filter((item) => item !== value)
    }

    if (values.length === 0) {
      values = ['all']
    }

    onChange(values)
  }

  return (
    <ScrollView
      horizontal
      style={{ flexGrow: 0 }}
      showsHorizontalScrollIndicator={false}
      accessibilityLabel="Specialty filters"
      accessibilityHint="Select specialties to filter doctors"
      accessibilityRole="list">
      <Chip
        marginLeft={24}
        onSelect={handleSelect}
        value="all"
        active={defaultValues.includes('all')}
        aria-label="All specialties chip"
        accessibilityHint="Toggles selection of all specialties filter"
        role="checkbox">
        All
      </Chip>
      {SPECIALTY_LIST.map(({ name, value }, index) => (
        <Chip
          key={value}
          onSelect={handleSelect}
          value={value}
          active={defaultValues.includes(value)}
          marginLeft={8}
          marginRight={index === SPECIALTY_LIST.length - 1 ? 24 : 0}
          aria-label={`${name} chip`}
          accessibilityHint={`Toggles selection of ${name.toLowerCase()} filter`}
          role="checkbox">
          {name}
        </Chip>
      ))}
    </ScrollView>
  )
}

export default MultipleSelectSpecialty
