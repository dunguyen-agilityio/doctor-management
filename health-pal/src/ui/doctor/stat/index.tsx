import { Text, XStack, YStack } from '@theme'

const DoctorStat = ({
  title,
  value,
  icon,
}: {
  title: string
  value: string | number
  icon: React.ReactNode
}) => {
  return (
    <YStack alignItems="center" gap={2}>
      <XStack
        justifyContent="center"
        alignItems="center"
        height={56}
        width={56}
        borderRadius={56}
        backgroundColor="$grey100">
        {icon}
      </XStack>
      <Text size="medium" fontWeight="600" color="$grey600">
        {value}
      </Text>
      <Text size="small" color="$grey500">
        {title}
      </Text>
    </YStack>
  )
}

export default DoctorStat
