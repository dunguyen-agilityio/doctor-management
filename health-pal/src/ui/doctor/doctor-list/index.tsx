import { FlashList, FlashListProps } from '@shopify/flash-list'

import { StyleSheet } from 'react-native'

import { Link } from 'expo-router'

import { View, XStack } from 'tamagui'

import { WINDOW_SIZE } from '@app/constants'

import useMediaQuery from '@app/hooks/use-media-query'

import { TDoctorData } from '@app/models/doctor'
import { formatDoctor } from '@app/utils/doctor'
import { keyExtractor } from '@app/utils/list'

import DoctorCard from '../doctor-card'

const ItemSeparatorComponent = () => <XStack height={8} />

interface DoctorListProps extends Omit<FlashListProps<TDoctorData>, 'renderItem'> {
  page?: number
}

const DoctorList = ({ page = 1, ...props }: DoctorListProps) => {
  const { height } = useMediaQuery({ height: 133, full: true })

  const Wrapper = ({
    id,
    children,
  }: React.PropsWithChildren<{
    id: string
  }>) => (
    <Link testID="doctor-link" href={{ pathname: '/doctors/details/[id]', params: { id } }}>
      {children}
    </Link>
  )

  const renderItem = ({ item }: { item: TDoctorData }) => {
    return <DoctorCard {...formatDoctor(item)} wrapper={Wrapper} />
  }

  return (
    <View minWidth={WINDOW_SIZE.width - 24 * 2} flex={1}>
      <FlashList
        renderItem={renderItem}
        contentContainerStyle={styles.contentContainerStyle}
        keyExtractor={keyExtractor}
        ItemSeparatorComponent={ItemSeparatorComponent}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        estimatedItemSize={height}
        {...props}
      />
    </View>
  )
}

export default DoctorList

const styles = StyleSheet.create({
  contentContainerStyle: { paddingHorizontal: 24, paddingVertical: 8 },
})
