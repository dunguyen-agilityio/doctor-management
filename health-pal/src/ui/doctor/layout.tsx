import React from 'react'

import { LoadingIndicator } from '@app/components'
import { DoctorContext } from '@app/contexts/doctor'
import useDoctors from '@app/hooks/use-doctors'

const DoctorLayout = ({ children }: React.PropsWithChildren) => {
  const { data, error, isLoading } = useDoctors()

  return (
    <>
      <DoctorContext value={data?.data ?? []}>{children}</DoctorContext>
      {isLoading && <LoadingIndicator />}
    </>
  )
}

export default DoctorLayout
