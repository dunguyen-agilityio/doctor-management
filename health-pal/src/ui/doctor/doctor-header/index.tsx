import { Button } from 'tamagui'

import { APP_TITLES } from '@app/constants/route'

import Header from '@app/components/header'
import useDoctor from '@app/hooks/use-doctor'
import { useFavoritesStore } from '@app/stores/favorite'
import { FAVORITE_TYPES } from '@app/types/favorite'
import { APP_ROUTES } from '@app/types/route'
import { FavoriteButton } from '@app/ui/favorite'

const DoctorDetailHeader = ({ id, goBack }: { id: string; goBack: () => void }) => {
  const { data, error, isLoading } = useDoctor(id)
  const doctorName = data?.name!
  const doctorId = data?.id!
  const favoriteId = useFavoritesStore((state) => state.favoriteDoctors[doctorId])

  const renderIcon = () => {
    if (isLoading || error || !data) return null

    return (
      <FavoriteButton
        favoriteId={favoriteId}
        itemId={doctorId}
        itemName={doctorName}
        type={FAVORITE_TYPES.DOCTOR}
        size={20}
        disabledStyle={{ backgroundColor: '$shadow1' }}
      />
    )
  }

  return (
    <Header title={APP_TITLES[APP_ROUTES.DOCTOR_DETAILS]} onBack={goBack}>
      <Button position="relative">{renderIcon()}</Button>
    </Header>
  )
}

export default DoctorDetailHeader
