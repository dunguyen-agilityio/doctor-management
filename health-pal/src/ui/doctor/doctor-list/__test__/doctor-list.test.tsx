import { MOCK_DOCTORS } from '@/mocks/doctor'
import { MOCK_USER } from '@/mocks/user'
import { render } from '@utils-test'

import DoctorList from '..'

jest.mock('@/hooks/use-require-auth', () => ({
  useRequireAuth: jest.fn().mockReturnValue({
    session: { user: MOCK_USER },
  }),
}))

describe('DoctorList', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot with empty data', () => {
    const { toJSON } = render(<DoctorList data={[]} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('matches snapshot with custom renderItem', () => {
    const { toJSON } = render(<DoctorList data={MOCK_DOCTORS} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
