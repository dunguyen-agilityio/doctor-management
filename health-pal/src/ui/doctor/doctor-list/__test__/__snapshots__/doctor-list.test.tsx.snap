// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorList matches snapshot with custom renderItem 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    style={
      {
        "flex": 1,
        "minWidth": 702,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        applyWindowCorrection={[Function]}
        canChangeSize={true}
        contentContainerStyle={
          {
            "backgroundColor": undefined,
            "minHeight": 1,
            "minWidth": 1,
            "paddingLeft": 24,
            "paddingRight": 24,
          }
        }
        contentHeight={0}
        contentWidth={0}
        data={
          [
            {
              "avatar": {
                "id": 1,
                "url": "https://picsum.photos/id/91/3504/2336",
              },
              "bio": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
              "clinic": {
                "address": "123 Main St, Springfield, USA",
                "description": "A premier healthcare facility offering top-tier services in Springfield.",
                "doctors": [],
                "documentId": "123e4567-e89b-12d3-a456-426614174000",
                "id": 1,
                "image": {
                  "alt": "Sunrise Hospital",
                  "id": 1,
                  "url": "https://picsum.photos/200",
                },
                "name": "Sunrise Hospital",
                "rating": 4.2,
                "reivewCouter": 300,
                "reviews": [],
                "type": "Clinic",
              },
              "dateOfBirth": 1980-01-01T00:00:00.000Z,
              "documentId": "doc_001",
              "email": "john.smith@example.com",
              "favoriteId": "fav_001",
              "gender": true,
              "id": 1,
              "name": "Dr. John Smith",
              "nickname": "jdoe",
              "rating": 4.8,
              "reviewCounter": 5,
              "reviews": [
                {
                  "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                  "documentId": "doc-1",
                  "id": 1,
                  "rating": 5,
                },
                {
                  "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                  "documentId": "doc-2",
                  "id": 2,
                  "image": {
                    "testUri": "../../../assets/images/user01.png",
                  },
                  "name": "Emily Anderson",
                  "rating": 5,
                },
                {
                  "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                  "documentId": "doc-3",
                  "id": 3,
                  "image": {
                    "testUri": "../../../assets/images/user01.png",
                  },
                  "name": "Emily Anderson",
                  "rating": 5,
                },
                {
                  "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                  "documentId": "doc-4",
                  "id": 4,
                  "image": {
                    "testUri": "../../../assets/images/user01.png",
                  },
                  "name": "Emily Anderson",
                  "rating": 5,
                },
              ],
              "specialty": {
                "description": "Specialized in heart-related conditions and treatments.",
                "doctors": [],
                "documentId": "123e4567-e89b-12d3-a456-426614174000",
                "id": 1,
                "name": "Cardiology",
              },
              "username": "jdoe",
              "users_permissions_user": {
                "avatar": {
                  "id": 1,
                  "url": "https://example.com/avatar.jpg",
                },
                "bio": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
                "dateOfBirth": 1990-01-01T00:00:00.000Z,
                "documentId": "123e4567-e89b-12d3-a456-426614174000",
                "email": "jdoe@example.com",
                "gender": true,
                "id": 1,
                "name": "John Doe",
                "nickname": "jdoe",
                "username": "jdoe",
              },
            },
          ]
        }
        dataProvider={
          DataProvider {
            "_data": [
              {
                "avatar": {
                  "id": 1,
                  "url": "https://picsum.photos/id/91/3504/2336",
                },
                "bio": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
                "clinic": {
                  "address": "123 Main St, Springfield, USA",
                  "description": "A premier healthcare facility offering top-tier services in Springfield.",
                  "doctors": [],
                  "documentId": "123e4567-e89b-12d3-a456-426614174000",
                  "id": 1,
                  "image": {
                    "alt": "Sunrise Hospital",
                    "id": 1,
                    "url": "https://picsum.photos/200",
                  },
                  "name": "Sunrise Hospital",
                  "rating": 4.2,
                  "reivewCouter": 300,
                  "reviews": [],
                  "type": "Clinic",
                },
                "dateOfBirth": 1980-01-01T00:00:00.000Z,
                "documentId": "doc_001",
                "email": "john.smith@example.com",
                "favoriteId": "fav_001",
                "gender": true,
                "id": 1,
                "name": "Dr. John Smith",
                "nickname": "jdoe",
                "rating": 4.8,
                "reviewCounter": 5,
                "reviews": [
                  {
                    "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                    "documentId": "doc-1",
                    "id": 1,
                    "rating": 5,
                  },
                  {
                    "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                    "documentId": "doc-2",
                    "id": 2,
                    "image": {
                      "testUri": "../../../assets/images/user01.png",
                    },
                    "name": "Emily Anderson",
                    "rating": 5,
                  },
                  {
                    "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                    "documentId": "doc-3",
                    "id": 3,
                    "image": {
                      "testUri": "../../../assets/images/user01.png",
                    },
                    "name": "Emily Anderson",
                    "rating": 5,
                  },
                  {
                    "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                    "documentId": "doc-4",
                    "id": 4,
                    "image": {
                      "testUri": "../../../assets/images/user01.png",
                    },
                    "name": "Emily Anderson",
                    "rating": 5,
                  },
                ],
                "specialty": {
                  "description": "Specialized in heart-related conditions and treatments.",
                  "doctors": [],
                  "documentId": "123e4567-e89b-12d3-a456-426614174000",
                  "id": 1,
                  "name": "Cardiology",
                },
                "username": "jdoe",
                "users_permissions_user": {
                  "avatar": {
                    "id": 1,
                    "url": "https://example.com/avatar.jpg",
                  },
                  "bio": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
                  "dateOfBirth": 1990-01-01T00:00:00.000Z,
                  "documentId": "123e4567-e89b-12d3-a456-426614174000",
                  "email": "jdoe@example.com",
                  "gender": true,
                  "id": 1,
                  "name": "John Doe",
                  "nickname": "jdoe",
                  "username": "jdoe",
                },
              },
            ],
            "_firstIndexToProcess": 0,
            "_hasStableIds": true,
            "_requiresDataChangeHandling": false,
            "_size": 1,
            "getStableId": [Function],
            "rowHasChanged": [Function],
          }
        }
        disableRecycling={false}
        estimatedItemSize={133}
        extendedState={{}}
        externalScrollView={[Function]}
        finalRenderAheadOffset={250}
        forceNonDeterministicRendering={true}
        horizontal={false}
        initialOffset={0}
        initialRenderIndex={0}
        isHorizontal={false}
        keyExtractor={[Function]}
        layoutProvider={
          GridLayoutProviderWithProps {
            "_acceptableRelayoutDelta": 1,
            "_getHeightOrWidth": [Function],
            "_getLayoutTypeForIndex": [Function],
            "_getSpan": [Function],
            "_hasExpired": false,
            "_maxSpan": 1,
            "_setLayoutForType": [Function],
            "_tempDim": {
              "height": 0,
              "width": 0,
            },
            "averageWindow": AverageWindow {
              "currentAverage": 133,
              "currentCount": 1,
              "inputValues": [
                133,
              ],
              "nextIndex": 1,
            },
            "defaultEstimatedItemSize": 100,
            "layoutObject": {
              "size": undefined,
              "span": undefined,
            },
            "props": {
              "ItemSeparatorComponent": [Function],
              "contentContainerStyle": {
                "paddingHorizontal": 24,
                "paddingVertical": 8,
              },
              "data": [
                {
                  "avatar": {
                    "id": 1,
                    "url": "https://picsum.photos/id/91/3504/2336",
                  },
                  "bio": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
                  "clinic": {
                    "address": "123 Main St, Springfield, USA",
                    "description": "A premier healthcare facility offering top-tier services in Springfield.",
                    "doctors": [],
                    "documentId": "123e4567-e89b-12d3-a456-426614174000",
                    "id": 1,
                    "image": {
                      "alt": "Sunrise Hospital",
                      "id": 1,
                      "url": "https://picsum.photos/200",
                    },
                    "name": "Sunrise Hospital",
                    "rating": 4.2,
                    "reivewCouter": 300,
                    "reviews": [],
                    "type": "Clinic",
                  },
                  "dateOfBirth": 1980-01-01T00:00:00.000Z,
                  "documentId": "doc_001",
                  "email": "john.smith@example.com",
                  "favoriteId": "fav_001",
                  "gender": true,
                  "id": 1,
                  "name": "Dr. John Smith",
                  "nickname": "jdoe",
                  "rating": 4.8,
                  "reviewCounter": 5,
                  "reviews": [
                    {
                      "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                      "documentId": "doc-1",
                      "id": 1,
                      "rating": 5,
                    },
                    {
                      "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                      "documentId": "doc-2",
                      "id": 2,
                      "image": {
                        "testUri": "../../../assets/images/user01.png",
                      },
                      "name": "Emily Anderson",
                      "rating": 5,
                    },
                    {
                      "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                      "documentId": "doc-3",
                      "id": 3,
                      "image": {
                        "testUri": "../../../assets/images/user01.png",
                      },
                      "name": "Emily Anderson",
                      "rating": 5,
                    },
                    {
                      "comment": "Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to",
                      "documentId": "doc-4",
                      "id": 4,
                      "image": {
                        "testUri": "../../../assets/images/user01.png",
                      },
                      "name": "Emily Anderson",
                      "rating": 5,
                    },
                  ],
                  "specialty": {
                    "description": "Specialized in heart-related conditions and treatments.",
                    "doctors": [],
                    "documentId": "123e4567-e89b-12d3-a456-426614174000",
                    "id": 1,
                    "name": "Cardiology",
                  },
                  "username": "jdoe",
                  "users_permissions_user": {
                    "avatar": {
                      "id": 1,
                      "url": "https://example.com/avatar.jpg",
                    },
                    "bio": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
                    "dateOfBirth": 1990-01-01T00:00:00.000Z,
                    "documentId": "123e4567-e89b-12d3-a456-426614174000",
                    "email": "jdoe@example.com",
                    "gender": true,
                    "id": 1,
                    "name": "John Doe",
                    "nickname": "jdoe",
                    "username": "jdoe",
                  },
                },
              ],
              "estimatedItemSize": 133,
              "keyExtractor": [Function],
              "numColumns": 1,
              "renderItem": [Function],
              "showsHorizontalScrollIndicator": false,
              "showsVerticalScrollIndicator": false,
            },
            "renderWindowInsets": {
              "height": 0,
              "width": -48,
            },
            "shouldRefreshWithAnchoring": true,
          }
        }
        maxRenderAhead={750}
        numColumns={1}
        onEndReached={[Function]}
        onEndReachedThreshold={0}
        onEndReachedThresholdRelative={0}
        onItemLayout={[Function]}
        onLayout={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onSizeChanged={[Function]}
        onVisibleIndicesChanged={[Function]}
        removeClippedSubviews={false}
        renderAheadOffset={0}
        renderAheadStep={250}
        renderContentContainer={[Function]}
        renderItem={[Function]}
        renderItemContainer={[Function]}
        rowRenderer={[Function]}
        scrollEventThrottle={16}
        scrollThrottle={16}
        scrollViewProps={
          {
            "contentContainerStyle": {
              "backgroundColor": undefined,
              "minHeight": 1,
              "minWidth": 1,
              "paddingLeft": 24,
              "paddingRight": 24,
            },
            "onLayout": [Function],
            "onScrollBeginDrag": [Function],
            "refreshControl": undefined,
            "style": {
              "minHeight": 1,
              "minWidth": 1,
            },
          }
        }
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        style={
          {
            "minHeight": 1,
            "minWidth": 1,
          }
        }
        suppressBoundedSizeException={true}
        windowCorrectionConfig={
          {
            "applyToInitialOffset": true,
            "applyToItemScroll": true,
            "value": {
              "endCorrection": 0,
              "startCorrection": 0,
              "windowShift": -0,
            },
          }
        }
      >
        <View>
          <View
            style={
              {
                "flexDirection": "column",
              }
            }
          >
            <AutoLayoutView
              enableInstrumentation={false}
              horizontal={false}
              onBlankAreaEvent={[Function]}
              onLayout={[Function]}
              renderAheadOffset={0}
              scrollOffset={0}
              style={
                {
                  "height": 0,
                  "width": 0,
                }
              }
              windowSize={0}
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>,
]
`;

exports[`DoctorList matches snapshot with empty data 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    style={
      {
        "flex": 1,
        "minWidth": 702,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        applyWindowCorrection={[Function]}
        canChangeSize={true}
        contentContainerStyle={
          {
            "backgroundColor": undefined,
            "minHeight": 1,
            "minWidth": 1,
            "paddingLeft": 24,
            "paddingRight": 24,
          }
        }
        contentHeight={0}
        contentWidth={0}
        data={[]}
        dataProvider={
          DataProvider {
            "_data": [],
            "_firstIndexToProcess": 0,
            "_hasStableIds": true,
            "_requiresDataChangeHandling": false,
            "_size": 0,
            "getStableId": [Function],
            "rowHasChanged": [Function],
          }
        }
        disableRecycling={false}
        estimatedItemSize={133}
        extendedState={{}}
        externalScrollView={[Function]}
        finalRenderAheadOffset={250}
        forceNonDeterministicRendering={true}
        horizontal={false}
        initialOffset={0}
        initialRenderIndex={0}
        isHorizontal={false}
        keyExtractor={[Function]}
        layoutProvider={
          GridLayoutProviderWithProps {
            "_acceptableRelayoutDelta": 1,
            "_getHeightOrWidth": [Function],
            "_getLayoutTypeForIndex": [Function],
            "_getSpan": [Function],
            "_hasExpired": false,
            "_maxSpan": 1,
            "_setLayoutForType": [Function],
            "_tempDim": {
              "height": 0,
              "width": 0,
            },
            "averageWindow": AverageWindow {
              "currentAverage": 133,
              "currentCount": 1,
              "inputValues": [
                133,
              ],
              "nextIndex": 1,
            },
            "defaultEstimatedItemSize": 100,
            "layoutObject": {
              "size": undefined,
              "span": undefined,
            },
            "props": {
              "ItemSeparatorComponent": [Function],
              "contentContainerStyle": {
                "paddingHorizontal": 24,
                "paddingVertical": 8,
              },
              "data": [],
              "estimatedItemSize": 133,
              "keyExtractor": [Function],
              "numColumns": 1,
              "renderItem": [Function],
              "showsHorizontalScrollIndicator": false,
              "showsVerticalScrollIndicator": false,
            },
            "renderWindowInsets": {
              "height": 0,
              "width": -48,
            },
            "shouldRefreshWithAnchoring": true,
          }
        }
        maxRenderAhead={750}
        numColumns={1}
        onEndReached={[Function]}
        onEndReachedThreshold={0}
        onEndReachedThresholdRelative={0}
        onItemLayout={[Function]}
        onLayout={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onSizeChanged={[Function]}
        onVisibleIndicesChanged={[Function]}
        removeClippedSubviews={false}
        renderAheadOffset={0}
        renderAheadStep={250}
        renderContentContainer={[Function]}
        renderItem={[Function]}
        renderItemContainer={[Function]}
        rowRenderer={[Function]}
        scrollEventThrottle={16}
        scrollThrottle={16}
        scrollViewProps={
          {
            "contentContainerStyle": {
              "backgroundColor": undefined,
              "minHeight": 1,
              "minWidth": 1,
              "paddingLeft": 24,
              "paddingRight": 24,
            },
            "onLayout": [Function],
            "onScrollBeginDrag": [Function],
            "refreshControl": undefined,
            "style": {
              "minHeight": 1,
              "minWidth": 1,
            },
          }
        }
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        style={
          {
            "minHeight": 1,
            "minWidth": 1,
          }
        }
        suppressBoundedSizeException={true}
        windowCorrectionConfig={
          {
            "applyToInitialOffset": true,
            "applyToItemScroll": true,
            "value": {
              "endCorrection": 0,
              "startCorrection": 0,
              "windowShift": -0,
            },
          }
        }
      >
        <View>
          <View
            style={
              {
                "flexDirection": "column",
              }
            }
          >
            <View
              style={
                {
                  "paddingLeft": 24,
                  "paddingTop": 8,
                }
              }
            />
            <View
              style={
                [
                  undefined,
                  undefined,
                ]
              }
            />
            <AutoLayoutView
              enableInstrumentation={false}
              horizontal={false}
              onBlankAreaEvent={[Function]}
              onLayout={[Function]}
              renderAheadOffset={0}
              scrollOffset={0}
              style={
                {
                  "height": 0,
                  "width": 0,
                }
              }
              windowSize={0}
            />
            <CellContainer
              index={-1}
              style={
                [
                  undefined,
                  undefined,
                ]
              }
            />
            <View
              style={
                {
                  "paddingBottom": 8,
                  "paddingRight": 24,
                }
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>,
]
`;
